commands/bootstrap/index.js

```diff
@@ -161,7 +161,7 @@ class BootstrapCommand extends Command {
     chain = chain.then(filteredPackages => {
       this.filteredPackages = filteredPackages;
 
-      if (filteredPackages.length !== this.targetGraph.size) {
+      if (filteredPackages.length !== this.targetGraph.size && !this.options.forceLocal) {
```

Comment:
Theoretically, this is what the `targetGraph` on line 168 was supposed to be doing:
```js
this.targetGraph = new PackageGraph(filteredPackages, "allDependencies", this.options.forceLocal);
```

I'm not sure this is the appropriate change.

Reply:
> Theoretically, this is what the targetGraph on line 168 was supposed to be doing:

Here is the problem: `filteredPackages` contains only packages in scope (as determined via `--scope` flag).

When I run the new test I have added:
- Original `targetGraph` contains entries for the following packages: `package-3`, `@test/package-1`, `package-2`, `package-4`
- `filteredPackages` contains entries for `package-4` only
- When `new PackageGraph` is called, it creates entries for `package-4` only. This looks reasonable to me, because we gave the constructor a list of packages containing `package-4` only.

Another problem I see in the current implementation: even when I provide `--force-local`, `lerna bootstrap` prints the warning _Installing local packages that do not match filters from registry_. That does not make sense to me - as I understand `--force-local`, it's purpose is to ensure that local packages are symlinked. So either `--force-local` is not doing what it advertises, or the warning is lying.

> I'm not sure this is the appropriate change.

I see.

Here is the problem I need to fix: when I run `lerna bootstrap --scope `package-4`, I want lerna to symlink all local packages (`@test/package-1`, `package-3`). This used to be the behavior before 71174e4 was landed and I want to get it back :)

Now my assumption is that `--force-local` is supposed to address my needs but does not work as intended right now. If that's right and you still think my change is not appropriate, then could you please advise me how to proceed to get the issue fixed? It's enough to give me pointers to other places where to look.

If my assumption is not correct and the current behavior of `--force-local` is intended, then could you please advise me what would be an acceptable solution/change in lerna that will give me:
 - all local dependencies installed as symlinks, regardless of `--scope` arguments
 - no warnings printed to console

Reply:
Conceptually:

The block `if (filteredPackages.length !== this.targetGraph.size) { /* use scoped targetGraph */ }` was introduced by 71174e4. 

In my change, I am skipping this block entirely when `options.forceLocal` is enabled, thus reverting back to behavior before 71174e4.